src/all/mangapark/src/eu/kanade/tachiyomi/extension/all/mangapark/MangaPark.kt

```diff
@@ -54,13 +54,31 @@ class MangaPark(
 
     private val apiUrl = "$baseUrl/apo/"
 
-    private val json: Json by injectLazy()
+    override val client = network.cloudflareClient.newBuilder().apply {
+        if (preference.getBoolean(ENABLE_NSFW, true)) {
+            addInterceptor(::siteSettingsInterceptor)
+            addNetworkInterceptor(CookieInterceptor(domain, "nsfw" to "2"))
+        }
+        rateLimitHost(apiUrl.toHttpUrl(), 1)
+        addInterceptor { chain ->
```

Comment:
Consider extracting to a class/function. Already done for `siteSettingsInterceptor`.